llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -982,6 +969,21 @@ PreservedAnalyses SpecConstantsPass::run(Module &M,
             generateSpecConstDefaultValueMetadata(DefaultValue));
       }
 
+      if (IsSYCLAlloca) {
+        // In case this is a 'sycl.llvm.alloca' intrinsic, use the emitted
+        // specialization constant as the allocation size.
+        auto *Intr = cast<SYCLAllocaInst>(CI);
+        // For emulation mode, use the default value for now. This code should
+        // never be run, as the runtime should throw a 'kernel_not_supported'
+        // exception.
```

Comment:
From my understanding if we have Level Zero/OpenCL AOT compilation scenario, then  we use emulation mode (HandlingMode::emulation) and in general case user may set new value for spec constant representing the size of alloca. As implemented in this PR - the new aspect is considered supported for Level Zero/OpenCL.
Could you please clarify where do  we throw kernel_not_supported error in such case? I.e. I don't understand where we differentiate between SPIRV/AOT case to throw such error?

Reply:
True. We should fail in that case. AOT compilation does not match with what we wanna achieve here. I added a check for emulated mode + SPIR-V target.

I will add an amendment to the spec and handle that in the frontend to report the error as soon as possible. Thanks!